### PR TITLE
[Doc] Update Customize Runtime Doc

### PR DESCRIPTION
**Pull request by MengzhangLI**

### CODE REVIEW

File: docs/en/advanced_guides/customize_runtime.md [hunk 1]
```diff
@@ -1,17 +1,217 @@
...
+  - [Optimizer](#optimizer)
+    - [Optimizer Wrapper](#optimizer-wrapper)
+    - [Constructor](#constructor)
+    - [Customize self-implemented optimizer](#customize-self-implemented-optimizer)
+    - [Customize optimizer constructor](#customize-optimizer-constructor)
+    - [Additional settings](#additional-settings)
+  - [Scheduler](#scheduler)
 
-### Customize optimizer supported by Pytorch
+In this tutorial, we will introduce some methods about how to customize runtime settings for the project.
```
Comment (MeowZheng) [inline docs/en/advanced_guides/customize_runtime.md:18]: I think it would be better to explain what runtime is in openmmlab

File: docs/en/advanced_guides/customize_runtime.md [hunk 2]
```diff
@@ -1,17 +1,217 @@
...
-We already support to use all the optimizers implemented by PyTorch, and the only modification is to change the `optimizer` field of config files.
-For example, if you want to use `ADAM` (note that the performance could drop a lot), the modification could be as the following.
+## Loop
+
+`Loop` means the workflow of training, validation or testing and we use `train_cfg`, `val_cfg` and `test_cfg` to build `Loop`.
+
+E.g.:
+
+```python
+# Use EpochBasedTrainLoop to train 200 epochs.
```
Comment (MeowZheng) [inline docs/en/advanced_guides/customize_runtime.md:27]:
```suggestion
# Use IterBasedTrainLoop to train 200 epochs.
```

File: docs/en/advanced_guides/customize_runtime.md [hunk 3]
```diff
@@ -1,17 +1,217 @@
...
+  - [Scheduler](#scheduler)
 
-### Customize optimizer supported by Pytorch
+In this tutorial, we will introduce some methods about how to customize runtime settings for the project.
 
-We already support to use all the optimizers implemented by PyTorch, and the only modification is to change the `optimizer` field of config files.
-For example, if you want to use `ADAM` (note that the performance could drop a lot), the modification could be as the following.
+## Loop
+
+`Loop` means the workflow of training, validation or testing and we use `train_cfg`, `val_cfg` and `test_cfg` to build `Loop`.
```
Comment (MeowZheng) [inline docs/en/advanced_guides/customize_runtime.md:22]: add 'MMEngine defines several [basic loops](https://github.com/open-mmlab/mmengine/blob/main/mmengine/runner/loops.py)'.

File: docs/en/advanced_guides/customize_runtime.md [hunk 4]
```diff
@@ -1,17 +1,217 @@
...
+    - [Step 2: Import the new hook](#step-2-import-the-new-hook)
+    - [Step 3: Modify the config](#step-3-modify-the-config)
+    - [Modify default runtime hooks](#modify-default-runtime-hooks)
+  - [Optimizer](#optimizer)
+    - [Optimizer Wrapper](#optimizer-wrapper)
+    - [Constructor](#constructor)
+    - [Customize self-implemented optimizer](#customize-self-implemented-optimizer)
+    - [Customize optimizer constructor](#customize-optimizer-constructor)
+    - [Additional settings](#additional-settings)
+  - [Scheduler](#scheduler)
```
Comment (MeowZheng) [inline docs/en/advanced_guides/customize_runtime.md:16]: it might need `toc` to build a table of contents

File: docs/en/advanced_guides/customize_runtime.md [hunk 5]
```diff
@@ -1,17 +1,217 @@
...
+# Use EpochBasedTrainLoop to train 200 epochs.
+train_cfg = dict(type='IterBasedTrainLoop', max_iters=80000, val_interval=8000)
+```
+
+MMEngine defines several [basic loops](https://github.com/open-mmlab/mmengine/blob/main/mmengine/runner/loops.py). Users could implement customized loops if the defined loops are not satisfied.
+
+## Hook
+
+Before learning to create your customized hooks, it is recommended to learn the basic concept of hooks in file [engine.md](https://github.com/open-mmlab/mmsegmentation/blob/dev-1.x/docs/en/advanced_guides/engine.md).
+
```
Comment (MeowZheng) [inline docs/en/advanced_guides/customize_runtime.md:36]: I think it might be better to introduce `defualt_hooks` first.

File: docs/en/advanced_guides/customize_runtime.md [hunk 6]
```diff
@@ -1,17 +1,217 @@
...
+```python
+logger = dict(
+    interval=50,
+    hooks=[
+        dict(type='TextLoggerHook'),
+        dict(type='TensorboardLoggerHook')
+    ])
+```
+
```
Comment (MeowZheng) [inline docs/en/advanced_guides/customize_runtime.md:142]: there is no TextLoggerHook TextLoggerHook WandbLoggerHook

File: docs/en/advanced_guides/customize_runtime.md [hunk 7]
```diff
@@ -1,17 +1,217 @@
...
+    sampler_seed=dict(type='DistSamplerSeedHook'),
+    visualization=dict(type='SegVisualizationHook'))
 ```
 
-To modify the learning rate of the model, the users only need to modify the `lr` in the config of optimizer. The users can directly set arguments following the [API doc](https://pytorch.org/docs/stable/optim.html?highlight=optim#module-torch.optim) of PyTorch.
+Here we reveals how what we can do with `logger` and `checkpoint`.
+
+#### Checkpoint config
+
+The MMCV runner will use `checkpoint_config` to initialize [`CheckpointHook`](https://github.com/open-mmlab/mmengine/blob/main/mmengine/hooks/checkpoint_hook.py#L19).
```
Comment (MeowZheng) [inline docs/en/advanced_guides/customize_runtime.md:121]: runner is in mmengine currently

Comment (MeowZheng) [inline docs/en/advanced_guides/customize_runtime.md:121]: why `checkpoint=dict(type='CheckpointHook', by_epoch=False, interval=2000)` in default hooks

but `checkpoint_config` is used to initialize `CheckpointHook`
